Simplenote/NoticeController.swift

```diff
@@ -0,0 +1,89 @@
+import UIKit
+
+class NoticeController {
+    static let shared = NoticeController()
+
+    private var notices: [Notice] = []
+    private var current: Notice?
+    private let noticePresenter = NoticePresenter()
+
+    private var activeViewIsBeingTouched: Bool = false
+
+    private init() { }
+
+    func setupNoticeController() {
+        noticePresenter.startListeningToKeyboardNotifications()
+    }
```

Comment:
Originally I did have them in the init, but I was having an issue with timing.  The Presenter wasn't being initialized by the controller until the first notice was created.  
IF the keyboard already was displayed when that happened, the presenter wouldn't know the current keyboard state so it would present the notices as if the keyboard wasn't there.
By moving start listening to it's own function, I could call it when the app was starting so it began listening right away.

Reply:
I think this can now be addressed?

Since the Presenter is **let** and `startListeningToKeyboardNotifications` only hooks up the keyboard notifications, (unless I'm missing something) you could now simplify things up by moving this initialization inside the `init()`

Reply:
@jleandroperez `static let shared = NoticeController()` is lazy, so we will still need to somehow access / instantiate it (and presenter along the way) when the app starts

Reply:
Please disregard my comment, I definitely missed the scenario in which the Keyboard would become visible, and the `NoticeController` would not start the Keyboard Notification listeners.

Sorry about the noise!! (and thank you @eshurakov !!)